examples/launch_job_template.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This example shows how to launch a job template.
+//
+// Use the following command to build and run it with all the debug output sent to the standard
+// error output:
+//
+//	go run launch_job_template.go \
+//		-url "https://awx.example.com/api" \
+//		-username "admin" \
+//      -proxy "http://proxy.com:3128" \
+//		-password "..." \
+//		-ca-file "ca.pem" \
+//		-logtostderr \
+//      -project "project" \
+//		-template "template-name" \
+//		-limit "node0.openshift.private" \
+// 		-extra-vars "alertname=bla job=my-job complex={\"simple\":\"label\"}" \
+//		-v=2
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	awx "github.com/moolitayer/awx-client-go/awx"
+)
+
+var (
+	url           string
+	username      string
+	password      string
+	proxy         string
+	insecure      bool
+	caFile        string
+	project       string
+	template      string
+	limit         string
+	extraVarsFlag string
+
+//	extraVar map[string]interface{}
+)
+
+func init() {
+	flag.StringVar(&url, "url", "https://awx.example.com/api", "API URL.")
+	flag.StringVar(&username, "username", "admin", "API user name.")
+	flag.StringVar(&password, "password", "password", "API user password.")
+	flag.StringVar(&proxy, "proxy", "", "API proxy URL.")
+	flag.BoolVar(&insecure, "insecure", false, "Don't verify server certificate.")
+	flag.StringVar(&caFile, "ca-file", "", "Trusted CA certificates.")
+	flag.StringVar(&project, "project", "", "Project Name.")
+	flag.StringVar(&template, "template", "", "Template Name.")
+	flag.StringVar(&limit, "limit", "", "Hosts limit")
+	flag.StringVar(&extraVarsFlag, "extra-vars", "", "extra variables to the Job")
+}
+
+func main() {
+	// Parse the command line:
+	flag.Parse()
+
+	extraVars, err := parseExtraVars(extraVarsFlag)
```

Comment:
I think from a user point of view this example would be most helpful if extraVars were set explicitly using literals.
I know this doesn't follow with the existing flow of taking everything from the commend line. You could use a literal in case `extra-vars` isn't provided.

Reply:
Done.
I also added an example for the ExtraVar API that allows setting a single variable